### PR TITLE
feat: add page descriptions and update titleType

### LINKED ISSUE #262: Update page titles & add page descriptions
Change the site template so that page metadata has
1. the variable part of the title first (so it is visible in small browser tabs) `Button - Carbon Design System`
2. an abstract/summary that matches the page url (e.g. summary of Button, from first paragraph) rather than the general site (summary of Carbon)

image

### PR DESCRIPTION
**Pull request by jnm2377**
Closes #262 

#### Changelog

**New**

- adds `description` to pages

**Changed**

- changes `titleType` to `prepend`

### CODE REVIEW

**Comment by jnm2377** [comment]
Hmm.. looks like something in a `ResourceCard` is making this fail though I'm unsure why. Some of the spacing got formatted but otherwise, I just added a description to the page. Looking into this.

File: src/pages/guidelines/accessibility/color.mdx [hunk 1]
```diff
@@ -20,7 +21,7 @@ tabs: ['Overview', 'Color', 'Keyboard', 'Developers']
 <ResourceCard
     subTitle="Carbon color contrast tool"
     href="https://marijohannessen.github.io/color-contrast-checker/"
-    
+
```
Comment (alisonjoseph) [inline src/pages/guidelines/accessibility/color.mdx]: build error seems to have something to do with this line break. I would try just updating it to

```
    <ResourceCard
      subTitle="Carbon color contrast tool"
      href="https://marijohannessen.github.io/color-contrast-checker/"
    />
```

File: src/pages/components/breadcrumb/code.mdx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Breadcrumb
+description: The breadcrumb component, or breadcrumb trail, is a secondary navigation pattern that shows hierarchy among content or traces a user’s path.
```
Comment (connor-leech) [inline src/pages/components/breadcrumb/code.mdx]:
```suggestion
description: The breadcrumb component is a secondary navigation pattern that shows hierarchy among content or traces a user’s path.
```

File: src/pages/components/breadcrumb/style.mdx [hunk 3]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Breadcrumb
+description: The breadcrumb component, or breadcrumb trail, is a secondary navigation pattern that shows hierarchy among content or traces a user’s path.
```
Comment (connor-leech) [inline src/pages/components/breadcrumb/style.mdx]:
```suggestion
description: The breadcrumb component is a secondary navigation pattern that shows hierarchy among content or traces a user’s path.
```

File: src/pages/components/breadcrumb/usage.mdx [hunk 4]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Breadcrumb
+description: The breadcrumb component, or breadcrumb trail, is a secondary navigation pattern that shows hierarchy among content or traces a user’s path.
```
Comment (connor-leech) [inline src/pages/components/breadcrumb/usage.mdx]:
```suggestion
description: The breadcrumb component is a secondary navigation pattern that shows hierarchy among content or traces a user’s path.
```

File: src/pages/components/button/code.mdx [hunk 5]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Button
+description: Buttons express what action will occur when the user clicks or touches it. Buttons are used to initialize an action, either in the background or foreground of an experience.
```
Comment (connor-leech) [inline src/pages/components/button/code.mdx]:
```suggestion
description: Buttons are used to initialize an action. Button labels express what action will occur when the user interacts with it.```

File: src/pages/components/button/style.mdx [hunk 6]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Button
+description: Buttons express what action will occur when the user clicks or touches it. Buttons are used to initialize an action, either in the background or foreground of an experience.
```
Comment (connor-leech) [inline src/pages/components/button/style.mdx]:
```suggestion
description: Buttons are used to initialize an action. Button labels express what action will occur when the user interacts with it.
```

File: src/pages/components/button/usage.mdx [hunk 7]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Button
+description: Buttons express what action will occur when the user clicks or touches it. Buttons are used to initialize an action, either in the background or foreground of an experience.
```
Comment (connor-leech) [inline src/pages/components/button/usage.mdx]:
```suggestion
description: Buttons are used to initialize an action. Button labels express what action will occur when the user interacts with it.
```

File: src/pages/components/content-switcher/code.mdx [hunk 8]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Content switcher
+description: Content switcher manipulates the content shown following an exclusive or “either/or” pattern. It is used to toggle between two or more content sections within the same space on screen.
```
Comment (connor-leech) [inline src/pages/components/content-switcher/code.mdx]:
```suggestion
description: Content switcher manipulates the content shown following an exclusive or “either/or” pattern. It is used to toggle between two or more content sections within the same space on screen.
```

File: src/pages/components/content-switcher/style.mdx [hunk 9]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Content switcher
+description: Content switcher manipulates the content shown following an exclusive or “either/or” pattern. It is used to toggle between two or more content sections within the same space on screen.
```
Comment (connor-leech) [inline src/pages/components/content-switcher/style.mdx]:
```suggestion
description: Content switchers manipulate the content shown following an exclusive or “either/or” pattern. It is used to toggle between two or more content sections within the same space on screen.
```

File: src/pages/components/content-switcher/usage.mdx [hunk 10]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Content switcher
+description: Content switcher manipulates the content shown following an exclusive or “either/or” pattern. It is used to toggle between two or more content sections within the same space on screen.
```
Comment (connor-leech) [inline src/pages/components/content-switcher/usage.mdx]:
```suggestion
description: Content switchers manipulate the content shown following an exclusive or “either/or” pattern. It is used to toggle between two or more content sections within the same space on screen.
```

File: src/pages/components/data-table/code.mdx [hunk 11]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Data table
+description: The data table component is designed in a way for additional functionality to be added on to fit your team's needs.
```
Comment (connor-leech) [inline src/pages/components/data-table/code.mdx]:
```suggestion
description: The data table component allows for the flexible display and sorting of information.
```

File: src/pages/components/data-table/style.mdx [hunk 12]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Data table
+description: The data table component is designed in a way for additional functionality to be added on to fit your team's needs.
```
Comment (connor-leech) [inline src/pages/components/data-table/style.mdx]:
```suggestion
description: The data table component allows for the flexible display and sorting of information.
```

File: src/pages/components/data-table/usage.mdx [hunk 13]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Data table
+description: The data table component is designed in a way for additional functionality to be added on to fit your team's needs.
```
Comment (connor-leech) [inline src/pages/components/data-table/usage.mdx]:
```suggestion
description: The data table component allows for the flexible display and sorting of information.
```

File: src/pages/components/file-uploader/code.mdx [hunk 14]
```diff
@@ -1,5 +1,6 @@
 ---
 title: File uploader
+description: File Uploader allows the user to transfer a file or submit content of their own.
```
Comment (connor-leech) [inline src/pages/components/file-uploader/code.mdx]:
```suggestion
description: File uploaders allow the user to transfer a file or submit content of their own.
```

File: src/pages/components/file-uploader/style.mdx [hunk 15]
```diff
@@ -1,5 +1,6 @@
 ---
 title: File uploader
+description: File Uploader allows the user to transfer a file or submit content of their own.
```
Comment (connor-leech) [inline src/pages/components/file-uploader/style.mdx]:
```suggestion
description: File uploaders allow the user to transfer a file or submit content of their own.
```

File: src/pages/components/file-uploader/usage.mdx [hunk 16]
```diff
@@ -1,5 +1,6 @@
 ---
 title: File uploader
+description: File Uploader allows the user to transfer a file or submit content of their own.
```
Comment (connor-leech) [inline src/pages/components/file-uploader/usage.mdx]:
```suggestion
description: File uploaders allow the user to transfer a file or submit content of their own.
```

File: src/pages/components/form/code.mdx [hunk 17]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Form
+description: Forms are used for submitting data so be as concise as possible when designing. Think about each field and what value the data will provide.
```
Comment (connor-leech) [inline src/pages/components/form/code.mdx]:
```suggestion
description: Forms are used for submitting data. Consider each field and what value the data will provide.
```

File: src/pages/components/form/style.mdx [hunk 18]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Form
+description: Forms are used for submitting data so be as concise as possible when designing. Think about each field and what value the data will provide.
```
Comment (connor-leech) [inline src/pages/components/form/style.mdx]:
```suggestion
description: Forms are used for submitting data. Consider each field and what value the data will provide.
```

File: src/pages/components/form/usage.mdx [hunk 19]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Form
+description: Forms are used for submitting data so be as concise as possible when designing. Think about each field and what value the data will provide.
```
Comment (connor-leech) [inline src/pages/components/form/usage.mdx]:
```suggestion
description: Forms are used for submitting data. Consider each field and what value the data will provide.
```

File: src/pages/components/inline-loading/code.mdx [hunk 20]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Inline loading
+description: Inline loading spinners are used when performing actions. They help notify user's that their action is being processed.
```
Comment (connor-leech) [inline src/pages/components/inline-loading/code.mdx]:
```suggestion
description: Inline loading spinners notify a user that their action is being processed.
```

File: src/pages/components/inline-loading/style.mdx [hunk 21]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Inline loading
+description: Inline loading spinners are used when performing actions. They help notify user's that their action is being processed.
```
Comment (connor-leech) [inline src/pages/components/inline-loading/style.mdx]:
```suggestion
description: Inline loading spinners notify a user that their action is being processed.
```

File: src/pages/components/inline-loading/usage.mdx [hunk 22]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Inline loading
+description: Inline loading spinners are used when performing actions. They help notify user's that their action is being processed.
```
Comment (connor-leech) [inline src/pages/components/inline-loading/usage.mdx]:
```suggestion
description: Inline loading spinners notify a user that their action is being processed.
```

File: src/pages/components/overflow-menu/code.mdx [hunk 23]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Overflow menu
+description: Overflow menu is used when additional options are available to the user and there is a space constraint.
```
Comment (connor-leech) [inline src/pages/components/overflow-menu/code.mdx]:
```suggestion
description: Overflow menus are used when additional options are available to the user and there is a space constraint.
```

File: src/pages/components/overflow-menu/style.mdx [hunk 24]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Overflow menu
+description: Overflow menu is used when additional options are available to the user and there is a space constraint.
```
Comment (connor-leech) [inline src/pages/components/overflow-menu/style.mdx]:
```suggestion
description: Overflow menus are used when additional options are available to the user and there is a space constraint.
```

File: src/pages/components/overflow-menu/usage.mdx [hunk 25]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Overflow menu
+description: Overflow menu is used when additional options are available to the user and there is a space constraint.
```
Comment (connor-leech) [inline src/pages/components/overflow-menu/usage.mdx]:
```suggestion
description: Overflow menus are used when additional options are available to the user and there is a space constraint.
```

File: src/pages/components/progress-indicator/code.mdx [hunk 26]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Progress indicator
+description: Progress indicator is a visual representation of a users progress through a set of steps. They guide the user through a number of steps in order to complete a specified process.
```
Comment (connor-leech) [inline src/pages/components/progress-indicator/code.mdx]:
```suggestion
description: A progress indicator is a visual representation of a user's progress through a set of steps. It guides the user through a number of steps in order to complete a specified process.
```

File: src/pages/components/progress-indicator/style.mdx [hunk 27]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Progress indicator
+description: Progress indicator is a visual representation of a users progress through a set of steps. They guide the user through a number of steps in order to complete a specified process.
```
Comment (connor-leech) [inline src/pages/components/progress-indicator/style.mdx]:
```suggestion
description: A progress indicator is a visual representation of a user's progress through a set of steps. It guides the user through a number of steps in order to complete a specified process.
```

File: src/pages/components/progress-indicator/usage.mdx [hunk 28]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Progress indicator
+description: Progress indicator is a visual representation of a users progress through a set of steps. They guide the user through a number of steps in order to complete a specified process.
```
Comment (connor-leech) [inline src/pages/components/progress-indicator/usage.mdx]:
```suggestion
description: A progress indicator is a visual representation of a user's progress through a set of steps. It guides the user through a number of steps in order to complete a specified process.
```

File: src/pages/components/radio-button/code.mdx [hunk 29]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Radio button
+description: Radio buttons are used when a list of two or more options are mutually exclusive, meaning the user must select only one option.
```
Comment (connor-leech) [inline src/pages/components/radio-button/code.mdx]:
```suggestion
description: Radio buttons are used when two or more options are mutually exclusive, meaning the user must select only one option.
```

File: src/pages/components/radio-button/style.mdx [hunk 30]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Radio button
+description: Radio buttons are used when a list of two or more options are mutually exclusive, meaning the user must select only one option.
```
Comment (connor-leech) [inline src/pages/components/radio-button/style.mdx]:
```suggestion
description: Radio buttons are used when two or more options are mutually exclusive, meaning the user must select only one option.
```

File: src/pages/components/radio-button/usage.mdx [hunk 31]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Radio button
+description: Radio buttons are used when a list of two or more options are mutually exclusive, meaning the user must select only one option.
```
Comment (connor-leech) [inline src/pages/components/radio-button/usage.mdx]:
```suggestion
description: Radio buttons are used when two or more options are mutually exclusive, meaning the user must select only one option.
```

File: src/pages/components/tile/code.mdx [hunk 32]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Tile
+description: Tiles are a highly flexible component for displaying a wide variety of content, including informational, getting started, how-to, next steps, and more.
```
Comment (connor-leech) [inline src/pages/components/tile/code.mdx]:
```suggestion
description: Tiles are a flexible component for displaying a wide variety of content, including informational, getting started, how-to, next steps, and more.
```

File: src/pages/components/tile/style.mdx [hunk 33]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Tile
+description: Tiles are a highly flexible component for displaying a wide variety of content, including informational, getting started, how-to, next steps, and more.
```
Comment (connor-leech) [inline src/pages/components/tile/style.mdx]:
```suggestion
description: Tiles are a flexible component for displaying a wide variety of content, including informational, getting started, how-to, next steps, and more.
```

File: src/pages/components/tile/usage.mdx [hunk 34]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Tile
+description: Tiles are a highly flexible component for displaying a wide variety of content, including informational, getting started, how-to, next steps, and more.
```
Comment (connor-leech) [inline src/pages/components/tile/usage.mdx]:
```suggestion
description: Tiles are a flexible component for displaying a wide variety of content, including informational, getting started, how-to, next steps, and more.
```

File: src/pages/components/toggle/code.mdx [hunk 35]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Toggle
+description: Toggle is a control that is used to quickly switch between two possible states. They are commonly used for “on/off” switches.
```
Comment (connor-leech) [inline src/pages/components/toggle/code.mdx]:
```suggestion
description: A toggle is a control that is used to quickly switch between two possible states. They are commonly used for “on/off” switches.
```

File: src/pages/components/toggle/style.mdx [hunk 36]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Toggle
+description: Toggle is a control that is used to quickly switch between two possible states. They are commonly used for “on/off” switches.
```
Comment (connor-leech) [inline src/pages/components/toggle/style.mdx]:
```suggestion
description: A toggle is a control that is used to quickly switch between two possible states. They are commonly used for “on/off” switches.
```

File: src/pages/components/toggle/usage.mdx [hunk 37]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Toggle
+description: Toggle is a control that is used to quickly switch between two possible states. They are commonly used for “on/off” switches.
```
Comment (connor-leech) [inline src/pages/components/toggle/usage.mdx]:
```suggestion
description: A toggle is a control that is used to quickly switch between two possible states. They are commonly used for “on/off” switches.
```

File: src/pages/contributions/bugs-and-requests.mdx [hunk 38]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Bugs and requests
+description: We use Github to track our bugs. If you have a bug to report or wish to request a new feature, please check our existing issues before opening a new one.
```
Comment (connor-leech) [inline src/pages/contributions/bugs-and-requests.mdx]:
```suggestion
description: Carbon tracks bugs and issues with GitHub. If you have a bug to report or wish to request a new feature, please check our existing issues before opening a new one.
```

File: src/pages/data-visualization/tooltip/code.md [hunk 39]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Tooltip
+description: Tooltips appear above or below a data representation depending on the use case. The quantitative value on the Tooltip corresponds to the numerical value that the data representation is depicting.
```
Comment (connor-leech) [inline src/pages/data-visualization/tooltip/code.md]:
```suggestion
description: Tooltips appear above or below a data representation depending on the use case. The quantitative value on the Tooltip corresponds to the numerical value that the data representation is depicting.
```

File: src/pages/data-visualization/tooltip/style.md [hunk 40]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Tooltip
+description: Tooltips appear above or below a data representation depending on the use case. The quantitative value on the Tooltip corresponds to the numerical value that the data representation is depicting.
```
Comment (connor-leech) [inline src/pages/data-visualization/tooltip/style.md]:
```suggestion
description: Tooltips appear above or below a data representation depending on the use case. The quantitative value on the Tooltip corresponds to the numerical value that the data representation is depicting.
```

File: src/pages/data-visualization/tooltip/usage.md [hunk 41]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Tooltip
+description: Tooltips appear above or below a data representation depending on the use case. The quantitative value on the Tooltip corresponds to the numerical value that the data representation is depicting.
```
Comment (connor-leech) [inline src/pages/data-visualization/tooltip/usage.md]:
```suggestion
description: Tooltips appear above or below a data representation depending on the use case. The quantitative value on the Tooltip corresponds to the numerical value that the data representation is depicting.
```

File: src/pages/get-started/about-carbon.mdx [hunk 42]
```diff
@@ -1,5 +1,6 @@
 ---
 title: About Carbon
+description: Carbon is IBM's open-source design system for digital products and experiences. With the IBM Design Language as its foundation, the system consists of working code, design tools and more.
```
Comment (connor-leech) [inline src/pages/get-started/about-carbon.mdx]:
```suggestion
description: Carbon is IBM's open-source design system for digital products and experiences. With the IBM Design Language as its foundation, the system consists of working code, design tools, and more.
```

File: src/pages/data-visualization/tooltip/style.md (hunk 40 above)
Comment (jnm2377) [reply src/pages/data-visualization/tooltip/style.md]: Not sure what the change is on this one?

**Comment by vpicone** [comment]
holy guacamole nice work!

File: src/pages/data-visualization/tooltip/style.md (hunk 40 above)
Comment (connor-leech) [reply src/pages/data-visualization/tooltip/style.md]: Neither am I 😂

**Pull request merged**